### PR TITLE
[FIX] Workflow task event reservation

### PR DESCRIPTION
**Pull request by rimbendhaou**
Action postpone task

### CODE REVIEW

File: project_event/views/project_task_view.xml [hunk 1]
```diff
@@ -127,10 +127,10 @@
         <field name="arch" type="xml">
             <form string="Task">
                 <header>
-                    <button name="action_option" string="Option" attrs="{'invisible': [('task_state', 'not in', 'draft')]}"
+                    <button name="action_option" string="Option" attrs="{'invisible': [('task_state', 'not in', ['draft', 'postponed'])]}"
```
Comment (eilst) [inline project_event/views/project_task_view.xml]: @wbeverly  is this correct?

File: project_event/views/project_task_view.xml [hunk 2]
```diff
@@ -127,10 +127,10 @@
...
                             type="object"
                             groups="project_event.group_project_event_editor"/>
-                    <button name="action_return_option" string="Return to Option" attrs="{'invisible': [('task_state', 'in', ['draft', 'option', 'done'])]}"
+                    <button name="action_return_option" string="Return to Option" attrs="{'invisible': [('task_state', 'in', ['draft', 'option', 'done', 'postponed'])]}"
```
Comment (eilst) [inline project_event/views/project_task_view.xml]: @wbeverly  is this correct?

File: project_event/views/project_task_view.xml [hunk 3]
```diff
@@ -7,9 +7,9 @@
         <field name="arch" type="xml">
             <form string="Activity">
                 <header>
-                    <button name="action_option" string="Option" attrs="{'invisible': [('task_state', 'not in', 'draft')]}"
+                    <button name="action_option" string="Option" attrs="{'invisible': [('task_state', 'not in', ['draft', 'postponed'])]}"
                             type="object"/>
-                    <button name="action_return_option" string="Return to Option" attrs="{'invisible': [('task_state', 'in', ['draft', 'option', 'done'])]}"
+                    <button name="action_return_option" string="Return to Option" attrs="{'invisible': [('task_state', 'in', ['draft', 'option', 'done', 'postponed'])]}"
```
Comment (eilst) [inline project_event/views/project_task_view.xml]: @wbeverly  is this correct?

File: project_event/views/project_task_view.xml [hunk 4]
```diff
@@ -7,9 +7,9 @@
         <field name="arch" type="xml">
             <form string="Activity">
                 <header>
-                    <button name="action_option" string="Option" attrs="{'invisible': [('task_state', 'not in', 'draft')]}"
+                    <button name="action_option" string="Option" attrs="{'invisible': [('task_state', 'not in', ['draft', 'postponed'])]}"
```
Comment (eilst) [inline project_event/views/project_task_view.xml]: @wbeverly  is this correct?

File: project_event/models/project_task.py [hunk 5]
```diff
@@ -862,8 +863,9 @@ def action_draft(self):
     @api.multi
     def action_postpone(self):
         if self.is_type_task() and self.task_state in \
-                ['requested', 'read', 'canceled', 'accepted']:
-            self.draft_resources_reservation()
+                ['requested', 'read', 'canceled', 'accepted', 'option']:
```
Comment (eilst) [inline project_event/models/project_task.py]: @wbeverly  is this correct?

Comment (eilst) [reply project_event/models/project_task.py]: @rimbendhaou @wbeverly I remember that the changes of the state were right, we tested them,  but maybe we missed this. Also, IMHO  a resource that is in state 'option', it doesn't make sense to postpone its event.

Comment (wbeverly) [reply project_event/models/project_task.py]: @eilst @rimbendhaou  You can refer to the document I left with rim, and also on : 
https://projects-odoo.savoirfairelinux.com/documents/23

You should have the possibility to 'REPORTER' an EV in OPTION, APPROVÉ
You should have the possibility to 'REPORTER' an AC in OPTION, APPROVÉ
You should have the possibility to 'REPORTER' an TA in OPTION, DEMANDÉ, LU, ACCEPTÉ

Comment (eilst) [inline project_event/models/project_task.py]:
```suggestion
                ['requested', 'read', 'canceled', 'accepted']:
```

File: project_event/models/project_task.py [hunk 6]
```diff
@@ -862,8 +863,9 @@ def action_draft(self):
     @api.multi
     def action_postpone(self):
         if self.is_type_task() and self.task_state in \
```
Comment (eilst) [inline project_event/models/project_task.py:861]: extract condition and name it something like check_send_message

File: project_event/models/project_task.py (hunk 5 above)
Comment (eilst) [reply project_event/models/project_task.py]: Right, but this condition if for managing messages

File: project_event/models/project_project.py [hunk 7]
```diff
@@ -51,7 +51,7 @@ class Project(models.Model):
         [
             ('draft', 'Draft'),
             ('option', 'Option'),
-            ('accepted', 'Accepted'),
+            ('accepted', 'Approved'),
```
Comment (eilst) [inline project_event/models/project_project.py]:
```suggestion
            ('approved', 'Approved'),
```

Comment (eilst) [reply project_event/models/project_project.py]: and replace all the following, to be consistent

File: project_event/models/project_project.py [hunk 8]
```diff
@@ -178,8 +174,7 @@ def get_message_body(action):
         return switcher.get(action)
 
     def get_message(self, action):
-        message = '<br>'
```
Comment (eilst) [inline project_event/models/project_project.py:182]: this break is for formatting

File: project_event/models/project_task.py [hunk 9]
```diff
@@ -779,6 +779,7 @@ def cancel_resources_reservation(self):
             reserve_event.write(
                 {'state': 'cancelled'}
             )
+            reserve_event.event_task_id = False
```
Comment (eilst) [inline project_event/models/project_task.py:782]: Clean states in field definition, these:  task_state = fields.Selection([
        ('draft', 'Draft'),
        ('option', 'Option'),
        ('requested', 'Requested'),
        ('read', 'Read'),
        ('postponed', 'Postponed'),
        ('accepted', 'Accepted'),
        ('approved', 'Approved'),
        ('done', 'Done'),
        ('canceled', 'Canceled')],
        string='State',
        default='draft',
        track_visibility='onchange',
    )
    task_state_report_done_required = fields.Selection([
        ('draft', 'Draft'),
        ('option', 'Option'),
        ('requested', 'Requested'),
        ('read', 'Read'),
        ('postponed', 'Postponed'),
        ('accepted', 'Accepted'),
        ('approved', 'Approved'),
        ('done', 'Done'),
        ('canceled', 'Canceled')],
        string='State',
        default='draft',
        compute='_compute_task_state_visible'
    )
    task_state_report_not_done_required = fields.Selection([
        ('draft', 'Draft'),
        ('option', 'Option'),
        ('requested', 'Requested'),
        ('read', 'Read'),
        ('postponed', 'Postponed'),
        ('accepted', 'Accepted'),
        ('approved', 'Approved'),
        ('done', 'Done'),
        ('canceled', 'Canceled')],
        string='State',
        default='draft',
        compute='_compute_task_state_visible'
    )

File: project_event/tests/test_project_event_reservation.py [hunk 10]
```diff
@@ -163,3 +54,49 @@ def test_060_check_partner_id_calendar_event_as_client_of_task(self):
             self.assertEqual(
                 reservation_event.client_type.id,
                 reservation_event.event_task_id.client_type.id)
+
+    def do_event_action_option(self, event):
```
Comment (eilst) [inline project_event/tests/test_project_event_reservation.py]: if these are tests, they should start with test_xxx

File: project_event/views/project_project_view.xml [hunk 11]
```diff
@@ -30,7 +30,7 @@
             <form string="Project Event">
             <header>
                     <button name="action_option" string="Option" attrs="{'invisible': [('state', '=', 'option')]}" type="object"/>
-                    <button name="action_accept" string="Accept" attrs="{'invisible': [('state', '=', 'accepted')]}" type="object"/>
+                    <button name="action_accept" string="Approve" attrs="{'invisible': [('state', '=', 'accepted')]}" type="object"/>
```
Comment (eilst) [inline project_event/views/project_project_view.xml]:
```suggestion
                    <button name="action_accept" string="Approve" attrs="{'invisible': [('state', '=', 'approved')]}" type="object"/>
```

**Review by eilst — APPROVED** [review]

**Pull request merged**